### DISCUSSION TITLE
Autoplaying video

### DISCUSSION THREAD

**Discussion by durraniu** [qmd-lab/closeread#132] [Q&A]
Hi, closeread is amazing! I made a small example as part of 30 day map challenge as shown below. 

Unfortunately, {mapgl} widgets are not yet supported in quarto, so I made video clips of flying to a location that {mapgl} generates. Initially, I set all videos to autoplay, but only the first video autoplays. I am not sure if that's a closeread issue or a quarto issue. Any ideas how to autoplay videos?

Source: https://github.com/durraniu/30_day_map_challenge_2024/tree/main/posts/day5

https://github.com/user-attachments/assets/97f18867-68c4-45dc-8291-d7babc98b06b

**Comment by andrewpbray** [comment]
Hello @durraniu , I'm so glad to see you're tackling this; mapbox integration is a long-term feature we'd like to add but for now, videos are a good way to handle this.

The way that stickies work is that *all* of the stickies in a section are stacked on top of one another in the main column of the page but they all have their opacity set to 0. When a particularly sticky is triggered, it's opacity changes to 1. I suspect that all of your videos are autoplaying right when the page loads but by the time you trigger them, they're already done.

This is related to #66 which @jimjam-slam has his eye on so I'll keep this discussion going over there for an eventual feature that we can add. A workaround to try in the meantime is to set up two cr-sections back to back. I haven't tried it, but it's possible this will delay the autoplay of the second video (note that the second video will scroll onto the screen instead of fade in, which probably isn't your intended transition).

**Comment by jimjam-slam** [comment]
Hey @durraniu, I'm getting to work on improving video support!

In the mean time, another option is to use OJS to initialise a Mapbox or MapLibre map. I'll try to put a demo of this in action, but the basic idea is that you use one OJS code block to initialise the map, and then another to tie Closeread scroll state to your map's state (ie. location/zoom). Hopefully I can demo this in the next few days!

**Reply by jimjam-slam** [reply]
Okay, I have a basic demo of using OJS with MapLibre on #140 (it's not on preview.closeread.dev yet, so if you'd like to see it in action, you'll have to clone the repo and run `quarto render` on the `docs` folder. But hopefully we'll have it online soon!)

**Comment by andrewpbray** [comment]
@durraniu Just FYI, we're working on general video support in #139 .
